vendor/bat-native-ledger/src/bat_get_media.cc

```diff
@@ -621,10 +715,49 @@ void BatGetMedia::onFetchPublisherFromDBResponse(ledger::Result result,
                                         _2,
                                         _3));
   } else {
+    if (providerType == TWITCH_MEDIA_TYPE) {
+      std::string twitchHandle = getUserFacingHandle(publisher_blob);
+      std::string twitchFaviconUrl =
+        getFaviconUrl(publisher_blob, twitchHandle);
+      if ((info->name != twitchHandle && !twitchHandle.empty()) ||
+          (info->favicon_url.empty() ||
+          info->favicon_url.find(".invalid") == std::string::npos)) {
+        updateAndSaveTwitchPublisher(
+          windowId,
+          twitchHandle,
+          twitchFaviconUrl,
+          result,
+          std::move(info),
+          visit_data,
+          providerType);
+      }
+    }
     ledger_->OnPublisherActivity(result, std::move(info), windowId);
   }
 }
 
+void BatGetMedia::updateAndSaveTwitchPublisher(
+    uint64_t windowId,
+    const std::string& twitchHandle,
+    const std::string& twitchFaviconUrl,
+    ledger::Result result,
+    std::unique_ptr<ledger::PublisherInfo> publisher_info,
+    const ledger::VisitData visit_data,
+    const std::string& providerType) {
+  ledger::VisitData new_visit_data(visit_data);
+  new_visit_data.name = twitchHandle;
+  new_visit_data.provider = providerType;
+  publisher_info->name = twitchHandle;
+  publisher_info->provider = providerType;
+  std::string favicon_key = "https://" + ledger_->GenerateGUID() +
+      ".invalid";
+  ledger_->FetchFavIcon(
```

Comment:
this needs to happen inside `saveVisitInternal` and we just need to set `visit_data.favicon_url` as  `twitchFaviconUrl`.

Reply:
Done.

Reply:
After further testing, there are cases where vods will not get processed correctly going this route. I would really like to push this and maybe address this in a separate PR.

Reply:
we can't fetch icons for unverified publishers, this is a blocker

Reply:
we release hotfix just to address this issue, so we can't reintroduce it, sorry. Will take a look and try to fix it to speed up the process

Reply:
Either way, using saveVisitInternal will prematurely increment scores, duration, etc. every time the panel is clicked. Will need further discussion.